Created jest-expo-enzyme (#5479)
expo/expo: packages/jest-expo-enzyme/README.md

import { View, StyleSheet, Platform } from 'react-native';

it(`renders a view with a custom background`, () => {
  const component = shallow(<View style={{ backgroundColor: 'rgba(0,0,0,0.5)' }} />);
  // To debug your component use this:
  console.log('Component:', component.debug({ verbose: true }));

  // When snapshot testing, you should always try and be as concise as possible
  // here we are extracting the style prop from `View` on native and `div` on web
  const prop = component.find(Platform.select({ default: 'View', web: 'div' })).prop('style');

  // Flatten the style so we can read it as an object
  const style = StyleSheet.flatten(prop);

  /**
   * Android: exports[`renders a view with a custom background 1`] = `"rgba(0,0,0,0.5)"`;
   * iOS: exports[`renders a view with a custom background 1`] = `"rgba(0,0,0,0.5)"`;
   * web: exports[`renders a view with a custom background 1`] = `"rgba(0,0,0,0.50)"`;
   */
  expect(style.backgroundColor).toMatchSnapshot();
});
```

## Customization

When constructing your tests you will more than likely want to build them one platform at a time. To do this you'll want to create a `jest.config.js` and mix platforms.

> This is done by adding single jest configs to the `projects` field of your custom Jest config, this is why method `withEnzyme` exists.

Reviewer: ```suggestion
This is done by specifying a separate Jest config in the `projects` field of your custom Jest config. The `withEnzyme` method's purpose is to let you augment each single-platform Jest config with the configuration fields needed to support Enzyme.

this is why method `withEnzyme` exists.
```

> You can use `withEnzyme` to add Enzyme support to an existing Jest config (**important** this is only officially supported with Jest configs provided by `jest-expo`, mixing with other configs may not work as expected). This implementation of Enzyme will also use a custom serializer for the React Native **StyleSheet** API which will make snapshot tests cleaner and easier to read.

Reviewer: ```suggestion
You can use `withEnzyme` to add Enzyme support to an existing Jest config (**important:** this is only officially supported with Jest configs provided by `jest-expo`; mixing with other configs may not work as expected).  also use a custom serializer for the React Native **StyleSheet** API, which will make snapshot tests cleaner and easier to read.
```

### Usage

> 1. Create a `jest.config.js` in your root directory and delete the `jest` field in your `package.json`. You can also create any `js` file and pass it to the Jest CLI with the `-c` flag, ex: `yarn jest -c custom.config.js`

Reviewer: ```suggestion
1. Create a `jest.config.js` in your root directory and delete the `jest` field in your `package.json`. You can also create any `js` file and pass it to the Jest CLI with the `-c` flag, ex: `yarn jest -c custom.config.js`.
```

> 2. Import the platforms you want to test and add enzyme support:

Reviewer: ```suggestion
2. Import the platforms you want to test and add Enzyme support:
```

```js

> // Skipping Node because components shouldn't be rendering in SSR

Reviewer: The point of SSR is to render components... rephrase
